File: Task5.1C/app/src/main/java/com/example/task51c/db/VideoDao.java
package com.example.task51c.db;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;
import com.example.task51c.model.VideoItem;
import java.util.List;

@Dao
public interface VideoDao {
    @Insert
    void insert(VideoItem video);

    @Query("SELECT * FROM VideoItem ORDER BY id DESC")
    List<VideoItem> getAll();
}

File: Task5.1C/app/src/main/java/com/example/itube/VideoDao.kt
package com.example.itube

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface VideoDao {
    @Query("SELECT * FROM videos")
    fun getAllVideos(): LiveData<List<Video>>

    @Insert
    suspend fun insertVideo(video: Video)

    @Delete
    suspend fun deleteVideo(video: Video)
} 